src/App.js

```diff
@@ -1,81 +1,112 @@
-import React from 'react';
-import {BrowserRouter as Router,
-  Switch,
+import React, {useEffect, useState} from 'react';
+import {
+  BrowserRouter as Router,
+  Redirect,
   Route,
-  Redirect} from 'react-router-dom';
+  Switch,
+} from 'react-router-dom';
 import './App.css';
 import AdminFrontPage from './components/admin/Admin';
 import InviteeForm from './components/form/InviteeForm';
 import LandingPage from './components/landingPage/LandingPage';
-import mockAuth from './components/Authenticator';
+import firebaseAuthenticator from './components/Authenticator';
 import StartingPage from './components/StartingPage';
 import SelfServicePage from './components/selfServicePage/SelfServicePage';
+import {useAuthUser, UserContext} from './firebaseFeatures';
 
-const PrivateRoute = ({authFunction, component, redirectPath}) => (
-  <Route
-    render={() => {
-      if (authFunction()) {
-        return component;
+function PrivateRoute({authFunction, component, redirectPath}) {
+  const [authorisationStatus, setAutorisationStatus] = useState(null);
+
+  useEffect(() => {
+    let cancelled = false;
+    authFunction().then((newValue) => {
+      if (!cancelled) {
+        setAutorisationStatus(newValue);
       }
-      return (<Redirect to={redirectPath}/>);
-    }} />
-);
+    });
+    return () => {
+      cancelled = true;
+    };
+  });
 
-function App() {
-  return (
-      <Router>
-        <div>
-          <Switch>
+  if (authorisationStatus === null) {
+    return <h1>Loading...</h1>;
```

Comment:
This briefly flashes during page transitions. Not a big deal, just slightly unpleasant to the eye.

Reply:
I already created ticket for that: https://github.com/googleinterns/step236-2020/issues/26